persist/fs/commitlog/iterator.go

```diff
@@ -185,6 +203,22 @@ func (i *iterator) nextReader() bool {
 	return true
 }
 
+func filterFiles(opts Options, files []string, predicate ReadEntryPredicate) ([]string, error) {
+	filteredFiles := []string{}
```

Comment:
Perhaps allocate this up front for single alloc?  Even though this is a low frequency called method, is still reasonable practice when you can very easily calculate the upper bound.

i.e.
```go
filteredFiles := make([]string, 0, len(files))
```